dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java

```diff
@@ -0,0 +1,87 @@
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.structure.factories.StructureFactory;
+import com.dotmarketing.portlets.structure.model.Relationship;
+import com.dotmarketing.portlets.structure.model.Structure;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class RelationshipAPITest extends IntegrationTestBase {
+
+    @BeforeClass
+    public static void prepare () throws Exception {
+
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+
+    }
+
+    /**
+     * Test to Attempt to create a Duplicated Relationship (the system should prevent it)
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void test_create_unique_relationships () throws DotDataException, DotSecurityException {
+
+        Relationship relationship = null;
+        Structure structure = null;
+
+        try {
+            structure = new Structure();
+            structure.setFixed(false);
+            structure.setVelocityVarName("IT-Structure-" + System.currentTimeMillis());
+            structure.setName("IT-Structure-"+ System.currentTimeMillis());
+            StructureFactory.saveStructure(structure);
+
+
+            final String relationTypeValue = "IT-Parent-Child" + System.currentTimeMillis();
+
+            relationship = new Relationship();
+            relationship.setParentRelationName("Parent");
+            relationship.setChildRelationName("Child");
+            relationship.setRelationTypeValue(relationTypeValue);
+            relationship.setParentStructureInode(structure.getInode());
+            relationship.setChildStructureInode(structure.getInode());
+
+            APILocator.getRelationshipAPI().save(relationship);
+            relationship = APILocator.getRelationshipAPI().byTypeValue(relationTypeValue);
+
+            Relationship duplicated = new Relationship();
+            duplicated.setParentRelationName("ParentDuplicated");
+            duplicated.setChildRelationName("ChildDuplicated");
+            duplicated.setRelationTypeValue(relationTypeValue);
+            duplicated.setParentStructureInode(structure.getInode());
+            duplicated.setChildStructureInode(structure.getInode());
+
+            DotDataException exception = null;
+            try {
+                APILocator.getRelationshipAPI().save(duplicated);
+            } catch (DotDataException ex) {
+                exception = ex;
+            }
+            //Assert an exception will be thrown because the relationship is duplicated
+            Assert.assertNotNull(exception);
+
+
+        } finally {
+            //This connection is useless after the Unique violation exception
+            DbConnectionFactory.closeConnection();
```

Comment:
Closing the connection is handled already by  `RelationshipAPI.save()` since it is annotated with `@WrapInTransaction`

Reply:
he is using a Factory here too